packages/react/src/components/UIShell/HeaderMenu.js

```diff
@@ -53,8 +47,13 @@ class HeaderMenu extends React.Component {
     tabIndex: PropTypes.number,
   };
 
+  static contextType = PrefixContext;
+  prefix = this.context;
```

Comment:
This is a new pattern to me for setting a class instance property based on context, is this a common pattern? I've always seen it used as `this.context` or if it needs a rename something like:

```js
const { context: prefix } = this;
```

My worry is that this would not change the value of `this.prefix` if the context value changes since this is only called when the class is initialized